larky/src/main/java/com/verygood/security/larky/modules/crypto/CryptoHashModule.java

```diff
@@ -55,12 +58,34 @@ public LarkyGeneralDigest SHA384() {
     return new LarkyGeneralDigest(digest);
   }
 
-  @StarlarkMethod(name = "SHA512")
-  public LarkyLongDigest SHA512() {
-    LongDigest digest = (LongDigest) DigestFactory.createSHA512();
+  @StarlarkMethod(name = "SHA512", parameters = {
+          @Param(name = "truncate", allowedTypes = {@ParamType(type = String.class)},
```

Comment:
what is this truncate flag?

Reply:
"The desired length of the digest. It can be either “224” or “256”. If not present, the digest is 512 bits long. "
Pycryptodome sha512 takes truncate as a parameter when init sha512 https://pycryptodome.readthedocs.io/en/latest/src/hash/sha512.html